libraries/standard/mqtt/utest/mqtt_utest.c

```diff
@@ -78,3 +142,546 @@ void test_MQTT_Init_Invalid_Params( void )
     mqttStatus = MQTT_Init( &context, &transport, &callbacks, NULL );
     TEST_ASSERT_EQUAL( MQTTBadParameter, mqttStatus );
 }
+
+/* ======================  Testing MQTT_ProcessLoop ========================= */
+
+/**
+ * @brief Initialize pNetworkBuffer using static buffer.
+ *
+ * @param[in] pNetworkBuffer Network buffer provided for the context.
+ */
+static void setupNetworkBuffer( MQTTFixedBuffer_t * const pNetworkBuffer )
+{
+    pNetworkBuffer->pBuffer = mqttBuffer;
+    pNetworkBuffer->size = MQTT_TEST_BUFFER_LENGTH;
+}
+
+/**
+ * @brief Mocked MQTT event callback.
+ */
+static void eventCallback( MQTTContext_t * pContext,
+                           MQTTPacketInfo_t * pPacketInfo,
+                           uint16_t packetIdentifier,
+                           MQTTPublishInfo_t * pPublishInfo )
+{
+    ( void ) pContext;
+    ( void ) pPacketInfo;
+    ( void ) packetIdentifier;
+    ( void ) pPublishInfo;
+}
+
+/**
+ * @brief A mocked timer query function that increments on every call. This
+ * guarantees that only a single iteration runs in the ProcessLoop for ease
+ * of testing.
+ */
+static uint32_t getTime( void )
+{
+    return globalEntryTime++;
+}
+
+/**
+ * @brief Mocked successful transport send.
+ */
+static int32_t transportSendSuccess( MQTTNetworkContext_t pContext,
+                                     const void * pBuffer,
+                                     size_t bytesToWrite )
+{
+    ( void ) pContext;
+    ( void ) pBuffer;
+    return bytesToWrite;
+}
+
+/**
+ * @brief Mocked successful transport read.
+ */
+static int32_t transportRecvSuccess( MQTTNetworkContext_t pContext,
+                                     void * pBuffer,
+                                     size_t bytesToRead )
+{
+    ( void ) pContext;
+    ( void ) pBuffer;
+    return bytesToRead;
+}
+
+/**
+ * @brief Initialize the transport interface with the mocked functions for
+ * send and receive.
+ */
+static void setupTransportInterface( MQTTTransportInterface_t * pTransport )
+{
+    pTransport->networkContext = 0;
```

Comment:
Are we missing coverage by always setting a NULL context?

Reply:
No, it's entirely up to the application to define the transport [receive](https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/development/libraries/standard/mqtt/include/mqtt_lightweight.h#L89) and [send](https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/development/libraries/standard/mqtt/include/mqtt.h#L40) functions, which are the only functions to directly use the network context

Reply:
I'm a little confused because `MQTTNetworkContext_t` is never passed as a parameter to the event callback, but I added 
```c
TEST_ASSERT_EQUAL( MQTT_SAMPLE_NETWORK_CONTEXT, pContext );
```
to the mocked transport send and receive if that's what you meant.